lib/net.c

```diff
@@ -1578,7 +1684,9 @@ finalize_stage2(__attribute__((unused)) void *arg)
 		if (ret < 0)
 			return ret;
 	}
-	return 0;
+	if (change_logs_permissions() < 0)
+		return -1;
```

Comment:
1. Make variable `ret` in previous block a local variable of `finalize_stage2()` and use `ret` here to receive the return of `change_logs_permissions()`, and return `ret` instead of `-1`.

2. If we change `gatekeeper_log_init()` to make it return the full path and file name of the log file, we can pass it to `finalize_stage2()` via `launch_at_stage2()`. This avoids scanning for log files in the folder and just update the log file created during initialization.

Reply:
On item 2, instead of making `gatekeeper_log_init()` return the full path and file name of the log file, make it return the file descriptor of the log file. Notice that no change to the prototype of `gatekeeper_log_init()` needs to be made in order to return the file descriptor because file descriptors are greater or equal to zero. Once `finalize_stage2()` receives the file descriptor, it can use `fchown(2)` to change the owner and group of the file.